Cargo.toml

```diff
@@ -48,7 +48,16 @@ dunce = "1.0.4"
 either = "1.13.0"
 esl01-renderdag = "0.3.0"
 futures = "0.3.30"
-git2 = "0.18.3"
+git2 = { version = "0.18.3", features = [
+    # Do *not* disable this feature even if you'd like dynamic linking. Instead,
+    # set the environment variable `LIBGIT2_NO_VENDOR=1` if dynamic linking must
```

Comment:
Should `LIBGIT2_NO_VENDOR=1` be set in `flake.nix` and in https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/version-management/jujutsu/default.nix after this PR? @thoughtpolice

Reply:
Good point! I tried to inform Homebrew packagers in #4080, I wonder whether there are other people to inform and/or better ways to inform them.

I checked, and indeed Nixpkg's `jj` is successfully dynamically linked to `libgit2`. I guess/hope Nix has no problem supporting multiple versions of libgit2 at once; it will need a different version after #4080.

Reply:
I could add a "Note to packagers" or "Breaking changes for packagers" to the changelog.

Reply:
This is now done. I think that's good enough. If #4080 gets in, we should mention that libgit2 0.18 is needed for dynamic linking as well; this would also affect Nixpkgs.

Reply:
@thoughtpolice or @necauqua may want to follow up with a change to `flake.nix`